### DISCUSSION TITLE
音源使用问题

### DISCUSSION THREAD

**Discussion by lyh20110** [1715173329/UnblockNeteaseMusic#629] [Q&A]
我在云服务器上使用exe文件进行部署，设置qq,kuwo,kugou三个音源，在听歌发现有以下问题：
1、使用qq音源，通过在Windows系统变量里设置QQ_Cookie后（已开通绿钻业务） 重启，可以听到320kbs的音质，但大概过了1天后，就只有获取到酷我的128kbs的了，qq的没有获取到了，必须要登录qq音乐官网获取新的QQ_Cookie中的qm_keyst的值，并重新设qm_keyst后才回到320kbs的音质，请问怎么解决？
2、kugou音源，用周董的歌曲七里香做测试，只有qq（在QQ_Cookie未过期时）和kuwo获取到音源连接，而kugou音源命令行提示以下的内容，这是怎么回事呢？
DEBUG: (cache) Something did not hit. Storing the execution result...
    logKey: "Something"
    cacheStorageId: "provider/kugou"
DEBUG: (request) Start requesting http://mobilecdn.kugou.com/api/v3/search/song?keyword=%E4%B8%83%E9%87%8C%E9%A6%99%20-%20%E5%91%A8%E6%9D%B0%E4%BC%A6&page=1&pagesize=10
DEBUG: (request) Start requesting http://www.kuwo.cn/api/www/search/searchMusicBykeyWord?key=%E4%B8%83%E9%87%8C%E9%A6%99%E5%91%A8%E6%9D%B0%E4%BC%A6&pn=1&rn=30
DEBUG: (request) Start requesting https://u.y.qq.com/cgi-bin/musicu.fcg?data=%7B%22req_0%22%3A%7B%22module%22%3A%22vkey.GetVkeyServer%22%2C%22method%22%3A%22CgiGetVkey%22%2C%22param%22%3A%7B%22guid%22%3A%225285990%22%2C%22loginflag%22%3A1%2C%22filename%22%3A%5B%22M8000012Ez0a1tFcOI.mp3%22%5D%2C%22songmid%22%3A%5B%22004Z8Ihr0JIu5s%22%5D%2C%22songtype%22%3A%5B0%5D%2C%22uin%22%3A%22529376719%22%2C%22platform%22%3A%2220%22%7D%7D%7D
DEBUG: (request) Start requesting https://u.y.qq.com/cgi-bin/musicu.fcg?data=%7B%22req_0%22%3A%7B%22module%22%3A%22vkey.GetVkeyServer%22%2C%22method%22%3A%22CgiGetVkey%22%2C%22param%22%3A%7B%22guid%22%3A%224885175%22%2C%22loginflag%22%3A1%2C%22filename%22%3A%5B%22M5000012Ez0a1tFcOI.mp3%22%5D%2C%22songmid%22%3A%5B%22004Z8Ihr0JIu5s%22%5D%2C%22songtype%22%3A%5B0%5D%2C%22uin%22%3A%22529376719%22%2C%22platform%22%3A%2220%22%7D%7D%7D
DEBUG: (request) Start requesting http://ws.stream.qqmusic.qq.com/M8000012Ez0a1tFcOI.mp3?guid=5285990&vkey=6D6B71896D48EEB7DD8FD44F1B9BA11EFDC01D0CE84E8125B3FDE4A788F1FF093D3381F4B326496E49FFC7A9A988124C982D0D7523DE6C38&uin=529376719&fromtag=66

**Comment by 1715173329** [comment]
QQ 音乐的 cookie 有效期仅有一天半，如有疑问请咨询马化腾。
酷狗也并非所有音乐都是免费的。

**Reply by joyoner** [reply]
> QQ 音乐的 cookie 有效期仅有一天半，如有疑问请咨询马化腾。
> 酷狗也并非所有音乐都是免费的。

如果是docker 环境变量也是一天半有效吗？

**Reply by pan93412** [reply]
是的，這部分是 QQ 音源伺服器的限制，不容易解決⋯⋯

**Reply by joyoner** [reply]
> 是的，這部分是 QQ 音源伺服器的限制，不容易解決⋯⋯

那这个音源是不是没啥实际意义，因为一天半就要手动重获取cookie，并重件docker，这样理解没错吧。

**Reply by pan93412** [reply]
有些人想要 QQ 音源，而且可以忍受這種麻煩。我自己是不用啦⋯⋯

**Reply by joyoner** [reply]
> 有些人想要 QQ 音源，而且可以忍受這種麻煩。我自己是不用啦⋯⋯

能不能储存账号密码定时刷新cookie…

**Reply by pan93412** [reply]
> > 有些人想要 QQ 音源，而且可以忍受這種麻煩。我自己是不用啦⋯⋯
> 
> 能不能储存账号密码定时刷新cookie…

有興趣的人可以實作然後發 PR ;)

**Reply by joyoner** [reply]
> 有些人想要 QQ 音源，而且可以忍受這種麻煩。我自己是不用啦⋯⋯

您自己用的音源及顺序是？或者UNM的默认音乐顺序是？
